integrations/sonarqube/client.py

```diff
@@ -157,13 +156,14 @@ async def get_components(
             logger.info(
                 f"Fetching all components in organization: {self.organization_id}"
             )
-        if api_query_params:
-            query_params.update(api_query_params)
-        else:
-            selector = cast(
-                SonarQubeProjectResourceConfig, event.resource_config
-            ).selector
-            query_params.update(selector.generate_request_params())
+        if self.is_onpremise:
+            if api_query_params:
+                query_params.update(api_query_params)
+            else:
+                selector = cast(
+                    SonarQubeProjectResourceConfig, event.resource_config
+                ).selector
+                query_params.update(selector.generate_request_params())
```

Comment:
lets detect if its not on premise and with api_query_params and raise a warning log